app/src/components/collectionStructure/accordionTree.tsx

```diff
@@ -0,0 +1,117 @@
+import { Box, Button, Collapse } from "@chakra-ui/react";
+import {
+  Flex,
+  Icon,
+  Text,
+  Tooltip,
+} from "@nypl/design-system-react-components";
+import { OpenStateItem } from "./collectionStructure";
+
+const ButtonText = ({
+  title,
+  hasSubContainers,
+  level,
+}: {
+  title: string;
+  hasSubContainers: boolean;
+  level: number;
+}) => {
+  const text = (
+    <Text
+      flex="1"
+      marginBottom="0"
+      whiteSpace="nowrap"
+      overflow="hidden"
+      textOverflow="ellipsis"
+      paddingLeft={hasSubContainers ? "s" : "28px"}
+      fontWeight="500"
+    >
+      {title}
+    </Text>
+  );
+
+  const truncationLength = 30 - level * 7;
+  return title.length > truncationLength ? (
+    <Tooltip zIndex="1000" content={title}>
+      {text}
+    </Tooltip>
+  ) : (
+    text
+  );
+};
+
+const AccordionTree = ({
```

Comment:
Minor but something to consider as you create components that will be used by others: add jsdoc style comments. I think it's important to highlight that this component must be used inside a `ul` or `ol` since it returns a list of `li`s and you can't tell that unless if you read and see how it's implemented.